packages/hoprd/crates/hoprd-hoprd/src/main.rs

```diff
@@ -0,0 +1,176 @@
+        info!("This is HOPRd {}", hopr_lib::constants::APP_VERSION);
+        
+        if std::env::var("DAPPNODE").map(|v| v.to_lowercase() == "true").unwrap_or(false) {
+            info!("The HOPRd node appears to run on DappNode");
+        }
+
+        let cfg = config::HoprdConfig::from_cli_args(args, false)?;
+        info!("Node configuration: {}", cfg.as_redacted_string()?);
+
+        // Find or create an identity
+        let identity_opts = IdentityOptions {
+            initialize: cfg.hopr.db.initialize,
+            id_path: cfg.identity.file.clone(),
+            password: cfg.identity.password.clone(),
+            use_weak_crypto: Some(cfg.test.use_weak_crypto),
+            private_key: cfg.identity.private_key.clone().and_then(|v| cli::parse_private_key(&v).ok()),
+        };
+
+        let hopr_keys = HoprKeys::init(identity_opts)?;
+
+        info!("This node has packet key '{}' and uses a blockchain address '{}'",
+            core_transport::Keypair::public(&hopr_keys.packet_key).to_peerid_str(),
+            core_transport::Keypair::public(&hopr_keys.chain_key).to_hex()
+        );
+
+        if core_transport::Keypair::public(&hopr_keys.packet_key).to_string().starts_with("0xff") {
+            warn!("This node uses an invalid packet key type and will not be able to become an effective relay node, please create a new identity!");
+        }
+
+        // Create the message inbox
+        let inbox: Arc<RwLock<hoprd_inbox::Inbox>> = Arc::new(RwLock::new(hoprd_inbox::inbox::MessageInbox::new_with_time(cfg.inbox.clone(), || {
+            std::time::Duration::from_millis(utils_misc::time::native::current_timestamp())
+        })));
+
+        let inbox_clone = inbox.clone();
+        let on_message = move |data: core_transport::ApplicationData| {
+            let now = Into::<DateTime<Utc>>::into(SystemTime::now()).to_rfc3339();
+            info!("#### NODE RECEIVED MESSAGE [{now}] ####");
+            let ingress = inbox_clone.clone();
+
+            //         let decodedMsg = decodeMessage(data.plain_text)
+            //         log(`Message: ${decodedMsg.msg}`)
+            //         log(`App tag: ${data.application_tag ?? 0}`)
+            //         log(`Latency: ${decodedMsg.latency} ms`)
+            //         metric_latency.observe(decodedMsg.latency)
+
+            //         if (RPCH_MESSAGE_REGEXP.test(decodedMsg.msg)) {
+            //           log(`RPCh: received message [${decodedMsg.msg}]`)
+            //         }
+
+            async_std::task::spawn_local(async move {
+                ingress.write().await.push(data).await
+            });
+        };
+
+        let on_acknowledgement = |_ack: HalfKeyChallenge| {
+            // TODO: needed by the websockets? 
+        };
+
+        // Create the node instance
+        info!("Creating the HOPRd node instance from hopr-lib");
+        // TODO: originally (DAPPNODE support) the safe and module address could have been undefined to allow safe setup
```

Comment:
If safe or module addr is not provided we should exit early, because hopr-lib cannot run without those.

Reply:
But what about the dappnode onboqrding procedure? That's what the comment likely referd to. @tolbrino @QYuQianchen

Reply:
The Dappnode won't start without those two values. User has to go through the on-boarding wizzard and then set those two values via ENV args and restart Hoprd. 

Previously, the node at least announced without Safe, but even this is not the case now.

Reply:
OK, one less TODO, thanks.